src/codegen.cpp

```diff
@@ -8894,6 +8894,13 @@ extern "C" void jl_init_llvm(void)
     if (jit_profiling && atoi(jit_profiling)) {
         jl_using_intel_jitevents = 1;
     }
+    if (!jl_using_intel_jitevents) {
+        // Check if we are running under Intel VTune Amplifier
+        const char *vtune = getenv("INTEL_JIT_PROFILER64");
+        if (vtune && strlen(vtune) > 0) {
+            jl_using_intel_jitevents = 1;
+        }
+    }
```

Comment:
Move this into a separate PR? It would also need to detect `INTEL_JIT_PROFILER32` I think.